### PR TITLE
Update PUDL to use frictionless v5

### PR DESCRIPTION
**Pull request by zaneselvans**
# Overview

* A step toward #3293 
* Updates dependencies to use Frictionless v5 which allows annotation of other tabular formats besides CSVs.
* Minor syntax changes where we're validating datapackage descriptors.
* Note that there's this related PR: https://github.com/catalyst-cooperative/pudl-archiver/pull/322

# Testing

* Ran `make pytest-coverage` locally in the new environment, using `ferc-xbrl-extractor>=1.4.0` and everything seemed to work.

```[tasklist]
# To-do list
- [ ] Update the release notes
- [ ] Ensure docs build, unit & integration tests, and test coverage pass locally with `make pytest-coverage`
- [ ] Review the PR yourself and call out any questions or issues you have
```

### CODE REVIEW

File: environments/conda-linux-64.lock.yml [hunk 1]
```diff
@@ -225,22 +226,21 @@ dependencies:
   - json-c=0.17=h7ab15ed_0
   - json5=0.9.25=pyhd8ed1ab_0
   - jsonpointer=2.4=py312h7900ff3_3
-  - jsonschema=4.21.1=pyhd8ed1ab_0
-  - jsonschema-specifications=2023.12.1=pyhd8ed1ab_0
-  - jsonschema-with-format-nongpl=4.21.1=pyhd8ed1ab_0
+  - jsonschema=4.17.3=pyhd8ed1ab_0
+  - jsonschema-with-format-nongpl=4.17.3=pyhd8ed1ab_0
```
Comment (zaneselvans) [inline environments/conda-linux-64.lock.yml:230]: Note that `frictionless` is currently pinning to `jsonschema<4.18` and `jupyter_events` also depends on that package, which results in a downgrade of some jupyter packages below. 

See https://github.com/frictionlessdata/frictionless-py/issues/1569

File: src/pudl/workspace/datastore.py [hunk 2]
```diff
@@ -171,12 +171,10 @@ def _validate_datapackage(self, datapackage_json: dict):
         Throws ValueError if invalid.
         """
         # TODO (daz): when we upgrade to frictionless>=5.0, this will be:
```
Comment (jdangerx) [inline src/pudl/workspace/datastore.py:173]:
```suggestion
```
Yes!! This TODO did its job!! Also it should be removed.

**Review by jdangerx — CHANGES_REQUESTED** [review]

**Review by jdangerx — APPROVED** [review]
I decided to just "commit suggestion" since there's no sense in going back-and-forth just to delete an obsolete TODO. Approved.

**Comment by zaneselvans** [comment]
TODID TODONE

**Comment by zaneselvans** [comment]
Silly stochastic unit test failure. Re-running.

**Pull request merged**